src/NLog/Config/ServiceRepositoryExtensions.cs

```diff
@@ -47,30 +47,50 @@ internal static ServiceRepository GetServiceProvider([CanBeNull] this LoggingCon
 
         internal static T ResolveService<T>(this ServiceRepository serviceProvider, bool ignoreExternalProvider = true) where T : class
         {
-            IServiceProvider externalServiceProvider;
-
-            try
+            if (ignoreExternalProvider)
             {
                 return serviceProvider.GetService<T>();
             }
-            catch (NLogDependencyResolveException)
+            else
             {
-                if (ignoreExternalProvider)
-                    throw;
+                IServiceProvider externalServiceProvider;
+
+                try
+                {
+                    var service = serviceProvider.TryGetService(typeof(T)) as T;
```

Comment:
generic method next, same as GetService, would be nice. 

(consistent API)

Reply:
Yes thank you for spotting. Fixed TryGetService to be generic method.